src/main/java/com/pinterest/secor/common/KafkaClient.java

```diff
@@ -141,8 +141,11 @@ private Message getMessage(TopicPartition topicPartition, long offset,
             payloadBytes = new byte[payload.limit()];
             payload.get(payloadBytes);
         }
+        Long timestamp = null;
+        timestamp = messageAndOffset.message().timestamp();
```

Comment:
Correct Henry, I have added the param `kafka.useKafkaTimestamp` in secor.common.properties and accordingly extracting timestamp, otherwise passing it as null. Please review and let me know if further changes required! Thank You.